### PR TITLE
[Doc] Added Inputs and Fields CSS Api documentation

### PR DESCRIPTION
**Pull request by WiXSL**

### CODE REVIEW

File: docs/Fields.md [hunk 1]
```diff
@@ -221,6 +221,14 @@ import { BooleanField } from 'react-admin';
 
 `<BooleanField>` also accepts the [common field props](./Fields.md#common-field-props).
 
+### CSS API
+
+| Rule name  | Description                 |
+| ---------- | --------------------------- |
+| `root`     | Applied to the root element |
+
+**Tip**: `"RaBooleanField"`: Global css name for overriding all instances of the component using Material UI theme. See their [documentation](https://material-ui.com/customization/globals/#css).
```
Comment (fzaninotto) [inline docs/Fields.md:230]:
```suggestion
To override the style of all instances of `<BooleanField>` using the [material-ui style overrides](https://material-ui.com/customization/globals/#css), use the `RaBooleanField` key.
```

File: docs/Fields.md [hunk 2]
```diff
@@ -545,11 +580,11 @@ import { SelectField } from 'react-admin';
...
-| `optionValue`     | Optional | `string`                      | 'id'    | Name of the field to compare to the value to find the matching choice                                                                        |
-| `translateChoice` | Optional | `Boolean`                     | true    | Whether or not the choice text should be translated                                                                                          |
+| Prop              | Required | Type                                         | Default | Description                                                                                                                                  |
+| ----------------- | -------- | -------------------------------------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------- |
+| `optionText`      | Optional | `string ` &#124; `Function` &#124; `Element` | 'name'  | Name of the field to use to display the matching choice, or function returning that field name, or a React element to render for that choice |
```
Comment (fzaninotto) [inline docs/Fields.md:585]: why did you change that? It renders fine in the doc site

File: docs/Fields.md [hunk 3]
```diff
@@ -545,11 +580,11 @@ import { SelectField } from 'react-admin';
...
-| ----------------- | -------- | ----------------------------- | ------- | -------------------------------------------------------------------------------------------------------------------------------------------- |
-| `optionText`      | Optional | `string | function | Element` | 'name'  | Name of the field to use to display the matching choice, or function returning that field name, or a React element to render for that choice |
-| `optionValue`     | Optional | `string`                      | 'id'    | Name of the field to compare to the value to find the matching choice                                                                        |
-| `translateChoice` | Optional | `Boolean`                     | true    | Whether or not the choice text should be translated                                                                                          |
+| Prop              | Required | Type                                         | Default | Description                                                                                                                                  |
```
Comment (fzaninotto) [inline docs/Fields.md:583]: Please refrain from reformatting tables like that - it's a never ending effort that has no real added value IMO.

File: docs/Fields.md [hunk 4]
```diff
@@ -639,15 +674,23 @@ With this configuration, `<ReferenceField>` wraps the user's name in a link to t
...
+| Prop        | Required | Type                       | Default  | Description                                                                                                         |
+| ----------- | -------- | -------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------- |
+| `reference` | Required | `string`                   | -        | The name of the resource for the referenced records, e.g. 'posts'                                                   |
+| `children`  | Required | `Element`                  | -        | The Field element used to render the referenced record                                                              |
+| `link`      | Optional | `string`                   | 'edit'   | Target of the link wrapping the rendered child                                                                      |
+| `sortBy`    | Optional | `string` &#124; `Function` | `source` | Name of the field to use for sorting when the user clicks on the column header. Set to `false` to disable the link. |
```
Comment (fzaninotto) [inline docs/Fields.md:682]: same

File: docs/Fields.md [hunk 5]
```diff
@@ -870,18 +913,26 @@ export const PostList = (props) => (
...
+| Prop         | Required | Type                       | Default                          | Description                                                                                                                                |
+| ------------ | -------- | -------------------------- | -------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------ |
+| `reference`  | Required | `string`                   | -                                | The name of the resource for the referenced records, e.g. 'tags'                                                                           |
+| `children`   | Required | `Element`                  | -                                | The Field element used to render the referenced records                                                                                    |
+| `sortBy`     | Optional | `string` &#124; `Function` | `source`                         | When used in a `List`, name of the field to use for sorting when the user clicks on the column header. Set to `false` to disable the link. |
```
Comment (fzaninotto) [inline docs/Fields.md:920]: same

**Review by fzaninotto — CHANGES_REQUESTED** [review]
Awesome work, it will really help developers. Looking forward to merging this one.

Could you please use the suggested wording for all fields / inputs and revert the reformatting of tables?

File: docs/Fields.md (hunk 2 above)
Comment (WiXSL) [reply docs/Fields.md]: I belive this kind of formatting is necesary (correctme if I'm wrong), That's my understanding of this commit b625602e21d98aa9af3db8194816dafecb4743e6 made by @djhi regarding  `jekyll` warnings being thrown

File: docs/Fields.md (hunk 3 above)
Comment (WiXSL) [reply docs/Fields.md:583]: Same

File: docs/Fields.md (hunk 4 above)
Comment (WiXSL) [reply docs/Fields.md:682]: Same

File: docs/Fields.md (hunk 5 above)
Comment (WiXSL) [reply docs/Fields.md:920]: Same

**Comment by WiXSL** [comment]
I've reverted the formatting of props table. I did'n get the problem was the &#124; formatting.

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
Thanks!
